kratos/processes/tetrahedral_mesh_orientation_check.cpp

```diff
@@ -26,42 +27,156 @@
 
 namespace Kratos
 {
-KRATOS_CREATE_LOCAL_FLAG(TetrahedralMeshOrientationCheck,ASSIGN_NEIGHBOUR_ELEMENTS_TO_CONDITIONS, 0);
-KRATOS_CREATE_LOCAL_FLAG(TetrahedralMeshOrientationCheck,COMPUTE_NODAL_NORMALS, 1);
-KRATOS_CREATE_LOCAL_FLAG(TetrahedralMeshOrientationCheck,COMPUTE_CONDITION_NORMALS, 2);
-KRATOS_CREATE_LOCAL_FLAG(TetrahedralMeshOrientationCheck,MAKE_VOLUMES_POSITIVE, 3);
-KRATOS_CREATE_LOCAL_FLAG(TetrahedralMeshOrientationCheck,ALLOW_REPEATED_CONDITIONS, 4);
+KRATOS_CREATE_LOCAL_FLAG(TetrahedralMeshOrientationCheck, THROW_ERRORS,                            0);
+KRATOS_CREATE_LOCAL_FLAG(TetrahedralMeshOrientationCheck, ASSIGN_NEIGHBOUR_ELEMENTS_TO_CONDITIONS, 1);
+KRATOS_CREATE_LOCAL_FLAG(TetrahedralMeshOrientationCheck, COMPUTE_NODAL_NORMALS,                   2);
+KRATOS_CREATE_LOCAL_FLAG(TetrahedralMeshOrientationCheck, COMPUTE_CONDITION_NORMALS,               3);
+KRATOS_CREATE_LOCAL_FLAG(TetrahedralMeshOrientationCheck, MAKE_VOLUMES_POSITIVE,                   4);
+KRATOS_CREATE_LOCAL_FLAG(TetrahedralMeshOrientationCheck, ALLOW_REPEATED_CONDITIONS,               5);
+KRATOS_CREATE_LOCAL_FLAG(TetrahedralMeshOrientationCheck, DETAILED_INVERTED_ENTITIES_MESSAGE,      6);
 
 /***********************************************************************************/
 /***********************************************************************************/
 
-typedef TetrahedralMeshOrientationCheck::SizeType SizeType;
+TetrahedralMeshOrientationCheck::TetrahedralMeshOrientationCheck(
+    ModelPart& rModelPart,
+    const bool ThrowErrors,
+    const Flags Options
+    ):  Process(),
+        mrModelPart(rModelPart),
+        mOptions(Options)
+
+{
+    // Set the THROW_ERRORS flag based on the input parameter
+    mOptions.Set(THROW_ERRORS, ThrowErrors);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+TetrahedralMeshOrientationCheck::TetrahedralMeshOrientationCheck(
+    ModelPart& rModelPart,
+    const Flags Options
+    ):  Process(),
+        mrModelPart(rModelPart),
+        mOptions(Options)
+{
+    // By default, do not throw errors unless specified
+    mOptions.Set(THROW_ERRORS, false);
+}
+
```

Comment:
I'd stick to the model-parameters ctors as this is to be the standard. Note that this must be eventually refactored to be an `Operation` (indeed it is something we can do easily), which is intended to only have model-parameters.